### DISCUSSION TITLE
Ötletek:[ csv import, hiányzó adatok]

### DISCUSSION THREAD

**Discussion by Netesfiu** [ZsBT/hass-w1000-portal#20] [Ideas]
**Sziasztok!**

Január óta tesztelem sikeresen az integrációt egy [másik hasonlóval](https://github.com/amargo/appdaemon-scripts), de momentán ez az integráció ami hosszútávon megbízhatóbb volt. Lehet nem csak nekem volt probléma az elején a felhasználó letiltása a sok bejelentkezés miatt, vagy a nevetségesen sűrű jelszómegújítás, ami okán kimaradtak napi adatok az energia fülön. Olyan kérdésem lenne, hogy az integrációba nem lehetne-e implementálni annak a lehetőségét, hogy manuális exportokat beolvastathassunk (pl egy csv, vagy xml exportot) hogy a kód végigellenőrizze, hogy hol vannak adateltérések/hiányok? Esetleg egy olyan rutint, hogy 2 hétre/hónapra visszanézi a nullás napokat? Jelenleg nem találtam semmi módját annak hogy, visszamenőlegesen pótoljam az adatokat ami szerintem több dolgot is orvosolna. (pl. áttelepítés, áramszünet, hálózati problémák stb. miatti adathiány)

Hozzá kell tegyem, hogy annyi programozási kompetenciám van, ami nagy nehezen elég template-ek összerakására a dashboardon. Ha erre lenne esetleg felhasználóbarát megoldás, akkor szeretnék kérni egy tutorialt

**Comment by ZsBT** [comment]
Szia!

A csv importot nem tervezem bevezetni, mert az integráció lényege az online
olvasás - függetlenül attól, hogy az EON adatok milyen rendszerességgel
érkeznek...
A visszadolgozással az a terv, hogy az első alkalommal 30 napra vissza
olvassa be az adatokat, azután pedig csak néhány napra. Tapasztalataim
szerint ha van portál adat elmaradás, az úgy két nap szokott lenni.

Üdvözlettel,
Zsombor

**Comment by ZsBT** [comment]
A jegyzőkönyv kedvéért; @Netesfiu volt kedves és készített importálót a visszamenő adatok betöltésére: https://github.com/Netesfiu/EON-SPOOKER

**Comment by missanhome** [comment]
Ha csak pár nap hiányzik (szokásos EON kimaradások miatt), arra találtam egy megoldást. Leírom, hátha valakinek segít.

@ZsBT, ez nyilván nem túl egészséges, mert a kódod szerkesztése, kódodban turkálás. Nem tudom mi a szokás, sajnos nem vagyok fejlesztő, kérlek szólj, ha ez aggályos vagy megsértek vele valamilyen szabályt, mert akkor törlöm a posztot azonnal.

Workaround:
A `/homeassistant/custom_components/w1000-energy-monitor/__init__.py` fájlban (jelenleg a 198-199. sorban) van két érték:

`        since = (now + timedelta(days=-2)).astimezone().strftime("%Y-%m-%dT23:00:00%z")`
`        until = (now + timedelta(days=0)).astimezone().strftime("%Y-%m-%dT%H:00:00%z")`

Ez a két érték határozza meg, hogy naponta hány napnyi adatot hozzon el a Portálról. A felső a kezdő (naptár szerint korábbi), az alsó a vég dátum (0 = ma). Ha ezt módosítod, elmented a fájlt és újraindítod a HA-t, akkor behúzza az új érték szerinti napi adatokat (visszamenőleg) is.

Tehát, ha mondjuk egy 4 nappal ezelőtti napi adat hiányzik és azt nem pótolta időben (2 napon belül) az EON, akkor átírod a since = (now + timedelta(days=-4))...-re és újraindítás után bent lesznek az adatok. Lehet játszani a since / until értékekkel pl. -10 / -6 a tíz és hat nappal ezelőtti intervallum közötti értékeket fogja behúzni.

Figyelem!
- Ha megvannak az adatok, mindig írd vissza az eredeti értékre, mentsd és indítsd újra a HA-t.
- Ne adj meg túl nagy időintervallumot (a két érték között túl sok nap ne legyen), mert valahol olvastam, hogy a masszív adat letöltés miatt zárolhatják a fiókodat. Én kb 5 napos egységekben húztam be egy teljes hónapot, azt még bírta.

**Reply by ZsBT** [reply]
Szia!
Egyáltalán nem aggályos a módosítás, sőt örülök ha más is nézi/írja a kódot. Valóban, a `since` és `until` változók határozzák meg a lekérdezés időszakát. Igazság szerint ezt a "-2" értéket ki kéne vezetni az integráció beállításaiba, hogy - akár felületről - is módosítani lehessen. Vagy, legalább a yaml konfigból vehetné az értéket, ahonnan a login nevet és jelszót.

**Comment by soosjanos88** [comment]
Sziasztok, 
Én is hasonlóra gondoltam hogy excelből importálja HA-ba. De én úgy gondoltam hogy fent van úgy is MariaDB ( mást miatt de annak még nem álltam neki) hogy oda ha beimportálom az excel tartalmát a táblába onnan , a táblából olvasná ki. 

Én mondjuk a GPT-t használtam hozzá mivel ilyen komoly tudásom nincs. 

**MariaDB + PHPMyAdmin + HA integráció**
https://chatgpt.com/share/687891b9-b964-800b-b4ea-45f648147ee0

import hiba 
https://chatgpt.com/share/687891de-fbfc-800b-88a1-7836105feb7e

Hátha esetleg hasznos lehet, ha nem akkor jelezzétek és törlöm a kommentet.
